### PR TITLE
Expectations fix

### LINKED ISSUE #21: [Feature Request] Patch regexp_instr for Trino, so that its compatible in dbt-expectations
Patch regexp_instr for Trino, so that its compatible in dbt-expectations

Reference, of why am here.
https://github.com/calogica/dbt-expectations/issues/243

Is your feature request related to a problem? Please describe.
Lets support regex expectation for Trino on
https://github.com/calogica/dbt-expectations/blob/b21b13f020f5df2c90885d0715a2e713d1204700/macros/regex/regexp_instr.sql

Describe the solution you'd like
We can use this,
https://trino.io/docs/current/functions/regexp.html
regexp_position(string, pattern, start, occurrence) → integer

Lets group similar implementation like https://github.com/dbt-msft/tsql-utils/tree/main/macros/dbt_expectations ?

### PR DESCRIPTION
**Pull request by wjhrdy**
This fixes #21 and one other bug when using the `dbt_expectations` package.

It adds the `dbt_expectations` test-suite which all runs without db errors but doesn't all pass weirdly. 

The two macros that have overrides are.

CleanShot 2023-10-07 at 00 39 48

Also added Readme section to help future contributors.

### CODE REVIEW

File: docker-compose-starburst.yml [hunk 1]
```diff
@@ -10,5 +10,5 @@ services:
 
 networks:
   default:
-    external:
-      name: dbt-net
+    external: true
```
Comment (hovaesco) [inline docker-compose-starburst.yml:13]: why? which docker version are you running?

File: README.md [hunk 2]
```diff
@@ -91,3 +91,63 @@ or for a dry run to preview dropped models:
 ```bash
 dbt run-operation trino__drop_old_relations --args "{dry_run: true}"
 ```
+
+## Contributing
```
Comment (hovaesco) [inline README.md]: I think we can move this section to a new file DEVELOPMENT.md

**Review by hovaesco — COMMENTED** [review]
(skimming)

**Comment by hovaesco** [comment]
Please rebase since there are some conflicts.

File: docker-compose-starburst.yml (hunk 1 above)
Comment (wjhrdy) [reply docker-compose-starburst.yml:13]: Docker Compose version 2.22.0 throws a warning about this being the new way to do it

**Comment by hovaesco** [comment]
Tests are not executed on the CI, please add them to `Makefile`.

**Comment by hovaesco** [comment]
something is failing in the CI https://github.com/starburstdata/dbt-trino-utils/actions/runs/6509366529/job/17732540173?pr=35#step:4:2696

**Comment by wjhrdy** [comment]
Do you think This is because I am importing the source project and also I copied the tests from the source project into the integration tests folder?

File: Makefile [hunk 3]
```diff
@@ -5,6 +5,7 @@ dbt-trino-tests:
 	./docker/run_dbt_utils_integration_tests.bash
 	./docker/run_dbt_date_integration_tests.bash
 	./docker/run_dbt_metrics_integration_tests.bash
+	./docker/run_dbt_expectations_integration_tests.bash
 	./docker/remove_trino.bash
 
 dbt-starburst-tests:
```
Comment (hovaesco) [inline Makefile:11]: please add also tests to `dbt-starburst-tests`

**Comment by hovaesco** [comment]
> Do you think This is because I am importing the source project and also I copied the tests from the source project into the integration tests folder?

Yes it might be this, it's some conflict between two `schema.yml`

**Comment by damian3031** [comment]
@wjhrdy Please pause your work on this PR. We are considering another approach, which involves adding Trino support directly to the `dbt-expectations` package rather than adding it to `dbt-trino-utils`. We believe this approach is better in terms of maintenance and is more user-friendly because users won't need to install an additional package or add extra configurations.

**Comment by wjhrdy** [comment]
That works!

Just for your reference, it was recommended by the `dbt-expectations` people to add this to `dbt-trino-utils`

https://github.com/calogica/dbt-expectations/issues/243#issuecomment-1441189188

**Comment by damian3031** [comment]
Trino is now supported directly in [dbt_expectations](https://hub.getdbt.com/calogica/dbt_expectations/latest/).
relevant PR: https://github.com/calogica/dbt-expectations/pull/294

**Pull request closed without merging**